pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -466,17 +455,13 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(no
 
 	for _, p := range configgen.Plugins {
 		chains := p.OnInboundFilterChains(pluginParams)
-		if len(chains) == 0 {
-			continue
-		}
-		if len(allChains) != 0 {
-			log.Warnf("Multiple plugins setup inbound filter chains for listener %s, FilterChainMatch may not work as intended!",
-				listenerMapKey)
-		}
 		allChains = append(allChains, chains...)
 	}
 	// Construct the default filter chain.
-	if len(allChains) == 0 {
+	if len(allChains) != 0 {
+		log.Debugf("Multiple plugins setup inbound filter chains for listener %s, FilterChainMatch may not work as intended!",
```

Comment:
this isn't the same behavior, before we checked `!= 0` before appending, now you check after appending. This will trigger anytime there is even one plugin

Reply:
The previous logs (`log.Warnf("Multiple plugins setup inbound filter chains for listener %s, FilterChainMatch may not work as intended!",	
				listenerMapKey)`) will not occur, this is not right. Except this , i do not change the append logic.